botorch/utils/sample_pareto.py

```
@@ -0,0 +1,281 @@
+    (iii) The genetic algorithm does not guarantee exactly `num_pareto_points` are
+        generated. Therefore, we apply a truncation strategy which selects
+        `num_greedy` points based on the sample hypervolume improvement, and then
+        selects the remaining points randomly.
+
+    Args:
+        model: The model. This does not support models which include fantasy
+            observations.
+        bounds: A `2 x d`-dim Tensor containing the input bounds.
+        num_pareto_samples: The number of GP samples.
+        num_pareto_points: The number of Pareto optimal points to be outputted.
+        maximize: If true we solve for the Pareto maximum.
+        num_generations: The number of generations of NSGA2.
+        pop_size: The population size maintained at each step of NSGA2.
+        num_offsprings: The number of offsprings used in NSGA2.
+        num_rff_features: The number of random Fourier features used for GP
+            sampling. Defaults to `512`.
+        max_tries: The maximum number of runs of NSGA2 to find num_pareto_points.
+        num_greedy: The number of points to select via the hypervolume improvement
+            truncation.
+        X_baseline: If `num_greedy > 0`, then we need to specify a `N x d`-dim Tensor
+            containing the training inputs used to perform the greedy selection.
+        ref_point: If `num_greedy > 0`, then we need to specify a `M`-dim Tensor
+            containing the reference point.
+
+    Returns:
+        A two-element tuple containing
+
+        - A `num_pareto_samples x num_pareto_points x d`-dim Tensor containing the
+            collection of Pareto optimal inputs.
+        - A `num_pareto_samples x num_pareto_points x M`-dim Tensor containing the
+            collection of Pareto optimal objectives.
+    """
+    tkwargs = {"dtype": bounds.dtype, "device": bounds.device}
+    M = model.num_outputs
+    d = bounds.shape[-1]
+
+    if M == 1:
```

It seems that this function should simply raise an exception if `M==1`.